packages/block-library/src/gallery/index.php

```diff
@@ -60,13 +60,11 @@ function block_core_gallery_render( $attributes, $content ) {
 		$gap = $gap && preg_match( '%[\\\(&=}]|/\*%', $gap ) ? null : $gap;
 	}
 
-	$class   = wp_unique_id( 'wp-block-gallery-' );
-	$content = preg_replace(
-		'/' . preg_quote( 'class="', '/' ) . '/',
-		'class="' . $class . ' ',
-		$content,
-		1
-	);
+	$class  = wp_unique_id( 'wp-block-gallery-' );
+	$walker = new WP_HTML_Walker( $content );
+	$walker->next_tag();
+	$walker->add_class( $class );
```

Comment:
might as well inline this code here since it's more clear now what's going on.

```php
$walker->add_class( wp_unique_id( 'wp-block-gallery-' ) );
```

??

Reply:
It's used later in one more place. That was my initial thought, too.

Reply:
I got confused by the original implementation and was about to suggest doing `set_attribute( 'class', $class );` 🤣  This is so much better!